java/jshell.support/src/org/netbeans/modules/jshell/tool/JShellTool.java

```diff
@@ -1691,50 +1754,60 @@ boolean cmdClasspath(String arg) {
     protected void classpathAdded(String s) {}
 
     boolean cmdDebug(String arg) {
-        if (arg.isEmpty()) {
-            debug = !debug;
-            InternalDebugControl.setDebugFlags(state, debug ? DBG_GEN : 0);
-            fluff("Debugging %s", debug ? "on" : "off");
-        } else {
-            int flags = 0;
-            for (char ch : arg.toCharArray()) {
-                switch (ch) {
-                    case '0':
-                        flags = 0;
-                        debug = false;
-                        fluff("Debugging off");
-                        break;
-                    case 'r':
-                        debug = true;
-                        fluff("REPL tool debugging on");
-                        break;
-                    case 'g':
-                        flags |= DBG_GEN;
-                        fluff("General debugging on");
-                        break;
-                    case 'f':
-                        flags |= DBG_FMGR;
-                        fluff("File manager debugging on");
-                        break;
-                    case 'c':
-                        flags |= DBG_COMPA;
-                        fluff("Completion analysis debugging on");
-                        break;
-                    case 'd':
-                        flags |= DBG_DEP;
-                        fluff("Dependency debugging on");
-                        break;
-                    case 'e':
-                        flags |= DBG_EVNT;
-                        fluff("Event debugging on");
-                        break;
-                    default:
-                        hard("Unknown debugging option: %c", ch);
-                        fluff("Use: 0 r g f c d");
-                        return false;
+        try {
+            if (arg.isEmpty()) {
+                debug = !debug;
+                MH_SET_DEBUG_FLAGS.invoke(state, debug ? DBG_GEN : 0);
```

Comment:
maybe guard against missing method ?

Reply:
Thanks for bringing this up. I added a null guard, that gives info to the user. While checking/testing this, I noticed, that my testing was tainted by modifications to the `netbeans.conf`. The package `jdk.internal.jshell.debug` needs to be exported to the unnamed module, that necessary change was also added.